torch/csrc/jit/script/init.cpp

```diff
@@ -83,6 +88,46 @@ struct VISIBILITY_HIDDEN PythonValue : public SugaredValue {
   py::object self;
 };
 
+// get the SugaredValue for something inside a torch.jit.Const
+// this can either be a ConstantPythonValue or a ModuleValue
+static std::shared_ptr<SugaredValue> createConstantSugaredValue(py::object obj);
+
+// by using torch.jit.Const, a user can mark a python value constant
+// we then make that value immutable.
+// once marked constant, we enable additional behavior such as
+// 1. conversion via asValue to a constant Tensor
+// 2. unrolling of for loops
+struct VISIBILITY_HIDDEN ConstantPythonValue : public PythonValue {
+  using PythonValue::PythonValue;
+  virtual Value * asValue(SourceRange loc, Method & m) override {
+    if(py::isinstance<py::int_>(self)) {
+      return createConstant(loc, m, at::CPU(at::kInt).scalarTensor(py::cast<int32_t>(self)));
+    } else if(py::isinstance<py::float_>(self)) {
+      return createConstant(loc, m, at::CPU(at::kFloat).scalarTensor(py::cast<float>(self)));
+    } else if(py::isinstance<py::bool_>(self)) {
+      return createConstant(loc, m, at::CPU(at::kByte).scalarTensor(py::cast<bool>(self)));
```

Comment:
We really should have a single function like `createBool`, `createFloat` to unify how we create primitive values in the IR. It will make maintenance much easier in case we'll ever want to change what we do